renderer/reducers/account.js

```diff
@@ -0,0 +1,266 @@
+import { send } from 'redux-electron-ipc'
+import { getIntl } from '@zap/i18n'
+import { sha256digest } from '@zap/utils/crypto'
+import createReducer from './utils/createReducer'
+import { settingsSelectors, saveConfigOverrides } from './settings'
+import { closeDialog } from './modal'
+import { showNotification } from './notification'
+import messages from './messages'
+
+// ------------------------------------
+// Initial State
+// ------------------------------------
+
+export const initialState = {
+  isAccountLoading: false,
+  isAccountLoaded: false,
+  initAccountError: null,
+  isLoggingIn: false,
+  loginError: null,
+  isLoggedIn: false,
+}
+
+// ------------------------------------
+// Constants
+// ------------------------------------
+
+export const INIT_ACCOUNT = 'INIT_ACCOUNT'
+export const INIT_ACCOUNT_SUCCESS = 'INIT_ACCOUNT_SUCCESS'
+export const INIT_ACCOUNT_FAILURE = 'INIT_ACCOUNT_FAILURE'
+
+export const LOGIN = 'LOGIN'
+export const LOGIN_SUCCESS = 'LOGIN_SUCCESS'
+export const LOGIN_FAILURE = 'LOGIN_FAILURE'
+export const LOGIN_CLEAR_ERROR = 'LOGIN_CLEAR_ERROR'
+
+export const CHANGE_PASSWORD_DIALOG_ID = 'CHANGE_PASSWORD_DIALOG'
+export const PASSWORD_PROMPT_DIALOG_ID = 'PASSWORD_PROMPT_DIALOG'
+export const PASSWORD_SET_DIALOG_ID = 'PASSWORD_SET_DIALOG_ID'
+
+// ------------------------------------
+// Actions
+// ------------------------------------
+
+/**
+ * initAccounts - Fetch the current account info from the database and save into the store.
+ * Should be called once when the app first loads.
+ *
+ * @returns {Function} Thunk
+ */
+export const initAccount = () => async (dispatch, getState) => {
+  dispatch({ type: INIT_ACCOUNT })
+  try {
+    // Auto login user if password feature is disabled.
+    const isAccountPasswordEnabled = accountSelectors.isAccountPasswordEnabled(getState())
+    if (!isAccountPasswordEnabled) {
+      dispatch({ type: LOGIN_SUCCESS })
+    }
+    dispatch({ type: INIT_ACCOUNT_SUCCESS })
+  } catch (error) {
+    dispatch({ type: INIT_ACCOUNT_FAILURE, error: error.message })
+  }
+}
+
+/**
+ * setPassword - Updates wallet password.
+ *
+ * @param {string} password new password
+ * @returns {Function} Thunk
+ */
+const setPassword = password => async dispatch => {
+  dispatch(send('setPassword', { password: await sha256digest(password) }))
```

Comment:
We should probably attach handlers for `setPasswordSuccess` and `setPasswordFailure` and resolve or reject this method accordingly based on the responses from those.